src/devices/src/virtio/net/device.rs

```diff
@@ -197,8 +193,8 @@ impl Net {
     }
 
     /// Provides the MAC of this net device.
-    pub fn guest_mac(&self) -> Option<&MacAddr> {
-        self.guest_mac.as_ref()
+    pub fn guest_mac(&self) -> MacAddr {
+        MacAddr::from_bytes_unchecked(&self.config_space.guest_mac[..MAC_ADDR_LEN])
```

Comment:
This fn seems to be called on the hot path. Why not cache the MacAddr when it's changed by the guest and always return a reference.

Reply:
Because we verify on the hotpath if frames source mac address is different than the interface assigned mac address (to update `METRICS.net.tx_spoofed_mac_count`), parsing the mac address from the config every time we process TX, would add few cycles to this processing step, and this can be avoided using a cache. However, every time the guest writes to the config space, we would need to update this cache accordingly. +1 to this suggestion.

Reply:
Done.